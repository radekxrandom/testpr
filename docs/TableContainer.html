<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: TableContainer</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: TableContainer</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>TableContainer<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description"><p>Container Component.
Contains logic for other components of the app.</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="TableContainer"><span class="type-signature"></span>new TableContainer<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="containers_TableContainer.js.html">containers/TableContainer.js</a>, <a href="containers_TableContainer.js.html#line14">line 14</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="addIncomesToData"><span class="type-signature"></span>addIncomesToData<span class="type-signature"></span></h4>




<div class="description">
    <p>Gets income data from the api,
calculates average, last month, and total income
for every company and adds those fields
to company object.</p>
</div>







    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>companysData</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;Object></span>


            
            </td>

            

            

            <td class="description last"><p>The objects Object[] w/ companys' info</p></td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="containers_TableContainer.js.html">containers/TableContainer.js</a>, <a href="containers_TableContainer.js.html#line37">line 37</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="cratePaginationTiles"><span class="type-signature"></span>cratePaginationTiles<span class="type-signature"></span></h4>




<div class="description">
    <p>Given companies data creates Object[] of objects containing
pagination tiles (ie buttons that allow user to navigate the table).
Tile is active when it corresponds to a currently displayed data.
variable numberOfRowsPerPage can be modified to change the number
of entries displayed on each page.</p>
</div>







    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>sortedCompanysdata</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;Object></span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="containers_TableContainer.js.html">containers/TableContainer.js</a>, <a href="containers_TableContainer.js.html#line70">line 70</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="onClickSortByField"><span class="type-signature"></span>onClickSortByField<span class="type-signature"></span></h4>




<div class="description">
    <p>Sort entries by a given field when it is clicked.
On second click reverse the order.
Arrow variable is passed as a prop to TableComponent and specifies
field in which arrow is displayed and it's orientation. Pointing downwards
if field is sorted ascending, and upwards otherwise.
After sorting it calls the method to paginate newly sorted data.</p>
</div>







    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>field</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Field by which data is to be sorted.</p></td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="containers_TableContainer.js.html">containers/TableContainer.js</a>, <a href="containers_TableContainer.js.html#line177">line 177</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="paginateData"><span class="type-signature"></span>paginateData<span class="type-signature"></span></h4>




<div class="description">
    <p>Generates rows with each company's data to be displayed on a currently
viewed page. Also switches relevant tiles object active property to true.</p>
</div>







    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>id</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last"><p>Current page number</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>sorted</code></td>
            

            <td class="type">
            
                
<span class="param-type">Boolean</span>


            
            </td>

            

            

            <td class="description last"><p>Whether data has been sorted or searched.</p></td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="containers_TableContainer.js.html">containers/TableContainer.js</a>, <a href="containers_TableContainer.js.html#line92">line 92</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="quickSort"><span class="type-signature"></span>quickSort<span class="type-signature"></span></h4>




<div class="description">
    <p>Sorts array with company objects by the given fields.</p>
</div>







    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>arr</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;Object></span>


            
            </td>

            

            

            <td class="description last"><p>The array to be sorted</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>field</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Field by which array is to be sorted</p></td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="containers_TableContainer.js.html">containers/TableContainer.js</a>, <a href="containers_TableContainer.js.html#line124">line 124</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="searchData"><span class="type-signature"></span>searchData<span class="type-signature"></span></h4>




<div class="description">
    <p>Searches data according to user input.
Afterwards call paginateData method.</p>
</div>







    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>e</code></td>
            

            <td class="type">
            
                
<span class="param-type">MyEvent</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="containers_TableContainer.js.html">containers/TableContainer.js</a>, <a href="containers_TableContainer.js.html#line202">line 202</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Render method</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="containers_TableContainer.js.html">containers/TableContainer.js</a>, <a href="containers_TableContainer.js.html#line241">line 241</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="PaginationNav.html">PaginationNav</a></li><li><a href="SearchComponent.html">SearchComponent</a></li><li><a href="TableComponent.html">TableComponent</a></li><li><a href="TableContainer.html">TableContainer</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.4</a> on Wed Apr 22 2020 05:35:58 GMT+0200 (Central European Summer Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>