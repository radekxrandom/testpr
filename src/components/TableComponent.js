import React from "react";

/**
 * Component used to display table with company information.
 * @component
 * @param {Object[]} displayedCompanies - Companies to display
 * @param {string} displayedCompanies[].name - Company name
 * @param {string} displayedCompanies[].city - Company city
 * @param {number} displayedCompanies[].avg - Company average income
 * @param {number} displayedCompanies[].total - Company total income
 * @param {number} displayedCompanies[].latest - Company last month's income
 * @param onClickSortByField - Function which sorts data by clicked field.
 */

const TableComponent = props => {
  return (
    <table>
      <thead>
        <tr id="col">
          <th onClick={() => props.onClickSortByField("id")}>
            <p className="controls">
              <span className="tblHdTxt">ID </span>
              {props.arrow === "id" && <span className="arrow"> &darr;</span>}
              {props.arrow === "idReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
          <th onClick={() => props.onClickSortByField("name")}>
            <p className="controls">
              <span className="tblHdTxt">NAME </span>
              {props.arrow === "name" && <span className="arrow"> &darr;</span>}
              {props.arrow === "nameReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
          <th onClick={() => props.onClickSortByField("city")}>
            <p className="controls">
              <span className="tblHdTxt">CITY </span>
              {props.arrow === "city" && <span className="arrow"> &darr;</span>}
              {props.arrow === "cityReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
          <th onClick={() => props.onClickSortByField("total")}>
            <p className="controls">
              <span className="tblHdTxt">TOTAL </span>
              {props.arrow === "total" && (
                <span className="arrow"> &darr;</span>
              )}
              {props.arrow === "totalReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
          <th onClick={() => props.onClickSortByField("avg")}>
            <p className="controls">
              <span className="tblHdTxt">AVERAGE </span>
              {props.arrow === "avg" && <span className="arrow"> &darr;</span>}
              {props.arrow === "avgReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
          <th onClick={() => props.onClickSortByField("latest")}>
            <p className="controls">
              <span className="tblHdTxt">LATEST </span>
              {props.arrow === "latest" && (
                <span className="arrow"> &darr;</span>
              )}
              {props.arrow === "latestReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
        </tr>
      </thead>
      <tbody>
        {props.displayedCompanies.length > 0 &&
          props.displayedCompanies.map(row => (
            <tr key={row.id}>
              <td>{row.id}</td>
              <td>{row.name}</td>
              <td>{row.city}</td>
              <td>{row.total}</td>
              <td>{row.avg}</td>
              <td>{row.latest}</td>
            </tr>
          ))}
        {!props.displayedCompanies.length && (
          <tr>
            <td align="center" className="tdSpanWholeRow" colspan="6">
              No search results
            </td>
          </tr>
        )}
      </tbody>
    </table>
  );
};

export default TableComponent;
