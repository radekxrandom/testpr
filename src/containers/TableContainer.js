import React, { Component } from "react";
import axios from "axios";
import TableComponent from "../components/TableComponent";
import PaginationNav from "../components/PaginationNav";
import SearchComponent from "../components/SearchComponent";

class TableContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [
        { id: "1", name: "naem", total: 12, avg: 5, last: 3 },
        { id: "2", name: "eams", total: 15, avg: 2, last: 4 }
      ],
      apiData: [],
      bool: false,
      tiles: [],
      showedFields: [],
      activePag: 0,
      arrow: "id",
      searchedPhrase: "",
      searchInput: "",
      loading: true
    };
  }

  asd = async arr => {
    return Promise.all(
      arr.map(async obj => {
        var objInc = await axios.get(
          `https://recruitment.hal.skygate.io/incomes/${obj.id}`
        );
        var values = objInc.data.incomes.map(obj => parseInt(obj.value));
        var totalInc = values.reduce((a, b) => a + b);
        var o3 = {
          total: totalInc,
          avg: totalInc / values.length,
          latest: values[values.length - 1]
        };
        return Object.assign(obj, o3);
      })
    );
  };

  paginate = id => {
    let showedFields = this.state.apiData.slice(id * 15, id * 15 + 15);
    let tiles = this.state.tiles.map(tile => {
      if (tile.id === id) {
        tile.active = true;
      }
      if (tile.id !== id) {
        tile.active = false;
      }
      return tile;
    });

    this.setState({
      showedFields,
      tiles,
      activePag: id,
      loading: false
    });
  };

  createTiles = sortedCompaniesData => {
    let tiles = [];
    for (let i = 0; i < sortedCompaniesData.length / 15; i++) {
      let tile = {
        clName: i === 1 ? "active" : "disabled",
        id: i
      };
      tiles.push(tile);
    }
    return tiles;
  };

  quickSort = (arr, field) => {
    if (arr.length <= 1) {
      return arr;
    } else {
      var right = [];
      var left = [];
      var pivot = arr.shift();

      var length = arr.length;

      for (let i = 0; i < length; i++) {
        if (arr[i][field] > pivot[field]) {
          right.push(arr[i]);
        } else {
          left.push(arr[i]);
        }
      }
      return [
        ...this.quickSort(left, field),
        pivot,
        ...this.quickSort(right, field)
      ];
    }
  };

  componentDidMount = async () => {
    const apiData = await axios.get(
      "https://recruitment.hal.skygate.io/companies"
    );
    console.log("got data");
    var sorted = this.quickSort(apiData.data, "id");
    console.log("sorted ok");
    var ugh = await this.asd(sorted);
    console.log("added monies");

    this.setState({
      apiData: ugh,
      tiles: this.createTiles(ugh)
    });
    this.paginate(0);
  };

  sortBy = async field => {
    if (this.state.arrow === field) {
      this.setState({
        apiData: this.state.apiData.reverse()
      });
    } else {
      await this.setState({
        apiData: this.quickSort(this.state.apiData, field),
        arrow: field
      });
    }

    this.paginate(this.state.activePag);
  };

  handleInput = async e => {
    this.setState({
      [e.target.name]: e.target.value
    });
    const srch = e.target.value;
    let result = this.state.apiData
      .filter(o => o.includes(srch.toLowerCase()))
      .sort(
        (a, b) => a.indexOf(srch.toLowerCase()) - b.indexOf(srch.toLowerCase())
      );
    await this.setState({
      apiData: result
    });
    this.paginate(this.state.activePag);
  };

  render() {
    if (this.state.loading) {
      return (
        <div>
          <div class="loader">Loading...</div>
        </div>
      );
    } else {
      return (
        <div>
          <SearchComponent handleInput={this.handleInput} />
          <TableComponent
            rowObjects={this.state.showedFields}
            sortBy={this.sortBy}
            arrow={this.state.arrow}
          />
          <PaginationNav tiles={this.state.tiles} clickedTile={this.paginate} />
        </div>
      );
    }
  }
}

export default TableContainer;
