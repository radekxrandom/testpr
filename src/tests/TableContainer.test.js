import React from "react";
import Enzyme, { shallow, mount, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TableContainer from "../containers/TableContainer";

//import child components
import TableComponent from "../components/TableComponent";

Enzyme.configure({ adapter: new Adapter() });

const mockData = [
  {
    avg: 5200.28,
    city: "North Shirleyfurt",
    id: 1,
    latest: 2652,
    name: "Ernser Group",
    total: 260014
  },
  {
    id: 2,
    name: "Franecki, Torphy and Lesch",
    city: "Port Halle",
    total: 251365,
    avg: 5027.3,
    latest: 7060
  },
  {
    id: 3,
    name: "Franecki, Torphy and Lesch",
    city: "Port Halle",
    total: 251365,
    avg: 5027.3,
    latest: 7060
  },
  {
    id: 4,
    name: "Smitham Group",
    city: "Rennerfort",
    total: 255396,
    avg: 5107.92,
    latest: 3167
  },
  {
    id: 6,
    name: "Towne - Vandervort",
    city: "Cassidyfurt",
    total: 283957,
    avg: 5679.14,
    latest: 1240
  }
];

describe("TblCont", () => {
  test("renders child comps", async () => {
    const wrapper = shallow(<TableContainer />);
    await wrapper.instance().componentDidMount();
    //wrapper.setState({ apiData: mockData, showedFields: mockData });
    const instance = wrapper.instance();
    instance.sortBy("id");
    expect(wrapper.state("apiData")).toBeTruthy();
    expect(wrapper.find(TableComponent).length).toEqual(1);
  });
});
