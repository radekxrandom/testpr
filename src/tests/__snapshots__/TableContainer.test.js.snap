// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TblCont renders ok and works ok 1`] = `ShallowWrapper {}`;
